sagemaker_train/launch_sagemaker_train.py

```diff
@@ -0,0 +1,198 @@
+    run_command(f"docker push {fullname}")
+    print("Sleeping for 5 seconds to ensure push succeeded")
+    time.sleep(5)
+
+    return f"{account}.dkr.ecr.{region}.amazonaws.com/{algorithm_name}:latest"
+
+
+def main():
+    # Use first line of file docstring as description if it exists.
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--build", action="store_true", help="Build image from scratch")
+    parser.add_argument("--update", action="store_true", help="Update code in image, don't re-build")
+    parser.add_argument("--local", action="store_true")
+    parser.add_argument("--user", required=True, help="User name")
+    parser.add_argument("--cfg-path", required=True, help="Launch config")
+
+    # AWS profile args
+    parser.add_argument("--region", default="us-east-1", help="AWS region")
+    parser.add_argument("--profile", default="poweruser", help="AWS profile to use")
+    parser.add_argument("--arn", default=None, help="If None, reads from SAGEMAKER_ARN env var")
+    parser.add_argument(
+        "--s3-remote-sync", default=None, help="S3 path to sync to. If none, reads from S3_REMOTE_SYNC env var"
+    )
+
+    # Instance args
+    parser.add_argument("--instance-count", default=1, type=int, help="Number of instances")
+    parser.add_argument("--instance-type", default="p4de", choices=list(INSTANCE_MAPPER.keys()))
+    parser.add_argument("--spot-instance", action="store_true")
+    args = parser.parse_args()
+
+    setup_tmp_name = "./setup_renamed_for_sagemaker.py"
+    # print(f"Renaming ./setup.py to {setup_tmp_name}")
+    # os.rename("./setup.py", setup_tmp_name)
+    try:
+        main_after_setup_move(args)
+    except:
+        # os.rename(setup_tmp_name, "./setup.py")
+        raise
+
+
+def main_after_setup_move(args):
+    image = get_image(
+        args.user,
+        args.region,
+        args.instance_type,
+        build_image=args.build,
+        update_image=args.update,
+        profile=args.profile,
+    )
+
+    ##########
+    # Create session and make sure of account and region
+    ##########
+    sagemaker_session = sagemaker.Session(boto_session=boto3.session.Session(region_name=args.region))
+
+    # provide a pre-existing role ARN as an alternative to creating a new role
+    role = get_arn(args.arn)
+    role_name = role.split(["/"][-1])
+    print(f"SageMaker Execution Role:{role}")
+    print(f"The name of the Execution role: {role_name[-1]}")
+
+    client = boto3.client("sts")
+    account = client.get_caller_identity()["Account"]
+    print(f"AWS account:{account}")
+
+    session = boto3.session.Session()
+    region = session.region_name
+    print(f"AWS region:{region}")
+
+    ##########
+    # Configure the training
+    ##########
+    base_job_name = f"{args.user.replace('.', '-')}-{NAME}"
+
+    checkpoint_local_path = "/opt/ml/checkpoints"
+
+    with open(args.cfg_path, "r") as f:
+        train_args = yaml.safe_load(f)
```

Comment:
Now that openlm supports a --config, let's just pass the config to openlm directly, with --config args.cfg_path?

Reply:
Tried this but it seems that it gives some typing errors when passed through sagemaker:
`Type mismatch (config: <class 'str'> vs. argparse: <class 'bool'>) with values (config:  vs. argparse: False) for config. key: dataset_resampled`

Leaving it as is for now